tests/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+use std::{
+  io::{BufRead, BufReader},
+  path::PathBuf,
+  process::{Child, Command, Stdio},
+  time::Duration,
+};
+
+use anyhow::{Context, Result};
+use tokio::time::sleep;
+
+struct TestSetup {
+  notary: Child,
+  client: Child,
+}
+
+impl TestSetup {
+  async fn new() -> Result<Self> {
+    // Find the workspace root directory
+    let workspace_root = {
+      let output = Command::new("cargo")
+        .args(["metadata", "--format-version", "1"])
+        .output()
+        .context("Failed to run cargo metadata")?;
+
+      if !output.status.success() {
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        anyhow::bail!("cargo metadata failed: {}", stderr);
+      }
+
+      let metadata: serde_json::Value =
+        serde_json::from_slice(&output.stdout).context("Failed to parse cargo metadata")?;
+
+      PathBuf::from(
+        metadata["workspace_root"].as_str().context("workspace_root not found in metadata")?,
+      )
+    };
+
+    println!("Workspace root: {:?}", workspace_root);
+
+    let notary_config = workspace_root.join("fixture/notary-config.toml");
+    let client_config = workspace_root.join("fixture/client.tee_tcp_local.json");
+
+    println!("Checking if config files exist:");
+    println!("Notary config exists: {}", notary_config.exists());
+    println!("Client config exists: {}", client_config.exists());
+
+    // Start notary
+    let notary = Command::new("cargo")
+      .args(["run", "-p", "notary", "--release", "--"])
+      .arg("--config")
+      .arg(&notary_config)
+      .env("RUST_LOG", "DEBUG")
+      .current_dir(&workspace_root)
+      .stdout(Stdio::inherit())
+      .stderr(Stdio::inherit())
+      .spawn()
+      .with_context(|| format!("Failed to spawn notary with config {:?}", notary_config))?;
+
+    sleep(Duration::from_secs(2)).await;
```

Comment:
I don't think 2 seconds are enough. @lonerapier has better data. I think we went with 30 seconds?

Reply:
Maybe we can wait for some specific output from the notary `.stdout`

Reply:
Yeah, let's do that.